arbnode/batch_poster.go

```diff
@@ -10,54 +10,61 @@ import (
 	"math/big"
 	"time"
 
+	"github.com/offchainlabs/nitro/arbnode/dataposter"
 	"github.com/offchainlabs/nitro/arbos"
 	"github.com/offchainlabs/nitro/util/headerreader"
 
 	"github.com/andybalholm/brotli"
 	"github.com/pkg/errors"
 	flag "github.com/spf13/pflag"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/ethereum/go-ethereum/params"
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/offchainlabs/nitro/arbstate"
 	"github.com/offchainlabs/nitro/arbutil"
 	"github.com/offchainlabs/nitro/das"
 	"github.com/offchainlabs/nitro/solgen/go/bridgegen"
-	"github.com/offchainlabs/nitro/util/arbmath"
 	"github.com/offchainlabs/nitro/util/stopwaiter"
 )
 
+type batchPosterPosition struct {
+	MessageCount        arbutil.MessageIndex
+	DelayedMessageCount uint64
+	NextSeqNum          uint64
+}
+
 type BatchPoster struct {
```

Comment:
maybe rename it to BatchBuilder?

Reply:
Hmm... Given that BatchPoster currently contains DataPoster, the BatchPoster is kind of the whole unit. The other issue with renaming it would be that we may want to rename a bunch of configs that all say batch poster.